src/board/__tests__/board.test.tsx

```diff
@@ -1,12 +1,31 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 import { KeyCode } from "@cloudscape-design/test-utils-core/utils";
-import { cleanup, render } from "@testing-library/react";
-import { afterEach, describe, expect, test } from "vitest";
+import { act, cleanup, render } from "@testing-library/react";
+import { afterEach, describe, expect, test, vi } from "vitest";
 import Board, { BoardProps } from "../../../lib/components/board";
+import boardStyles from "../../../lib/components/board/styles.css.js";
 import BoardItem from "../../../lib/components/board-item";
+import { mockController } from "../../../lib/components/internal/dnd-controller/__mocks__/controller";
+import { DragAndDropData } from "../../../lib/components/internal/dnd-controller/controller";
+import { Coordinates } from "../../../lib/components/internal/utils/coordinates";
 import createWrapper from "../../../lib/components/test-utils/dom";
 
+afterEach(cleanup);
+
+vi.mock("../../../lib/components/internal/dnd-controller/controller");
```

Comment:
Other tests in this file relied on genuine instance of the controller. Can we avoid this mock and use mousedown/mouseup events instead?

(probably, it should be a public test-util too, but let's do manual events right here for now)

Reply:
I was struggling to get this to work with `fireEvent` or `.click()`. It wasn't updating my rendered component correctly. Any idea what I might be doing wrong?